lib/logflare/backends/source_sup_worker.ex

```diff
@@ -33,7 +33,7 @@ defmodule Logflare.Backends.SourceSupWorker do
 
   defp do_check(source) do
     backends = Backends.list_backends(source_id: source.id)
-    rules = Rules.list_rules(source)
+    rules = Rules.Cache.list_rules(source)
```

Comment:
might be good to do a Stream in the future, so that we don't hold too many rules in the cache.